### ISSUE TITLE
Feedback about GeneratedMapper

### ISSUE THREAD

**Issue by ignatandrei** [ThomasBleijendaal/GeneratedMapper#11]
Hello Thomas Bleijendaal

I have started a mini-e-book about Roslyn Source Code Generators(RSCG).

Your RSCG , GeneratedMapper from https://github.com/ThomasBleijendaal/GeneratedMapper , is between them at https://github.com/ignatandrei/RSCG_Examples/tree/main/DTOMapper

I am thinking to add to the e-book the some more information regarding the RSCG from the authors of RSCG.

So could you please answer to the following :

1. Short info about you , Thomas Bleijendaal

2. Why did you start this GeneratedMapper  ?

3. How do yourself use your GeneratedMapper ?

4. What other RSCG do you use ? 

5. Any other feedback ?

Thanks

Andrei Ignat
http://msprogrammer.serviciipeweb.ro/

**Comment by ThomasBleijendaal** [comment]
Hi @ignatandrei,

Cool book!

To answer your questions: I'm Thomas Bleijendaal, I'm a .NET developer working for Triple in Alkmaar, The Netherlands. I've started GeneratedMapper to get more familiar with source generation. While the tooling is a bit rough now, I do believe that this feature can really bring a lot of value to .NET. 

I've been burned before by a lot of hard-to-find bugs caused by badly behaving mappers and runtime surprises from badly handled null-references. I wanted to create a object-to-object mapper that would be very picky and raise compilation errors instead of runtime exceptions. Being able to see what the mapper will do by simply inspecting the code is very handy, and makes you trust your mapper even more.

I use the GeneratedMapper in a project where I map models coming from Contentful to DTOs that are used in views. I've also tried to use it in another project, but because that targeted an older runtime I could not make that work. But, since I could copy all the generated mappers and maintain those manually, switching away from it wasn't that bad. I think that is also very valuable of code generation, you still have a copy of what a generator made for you.

I haven't really used any other RSCGs yet, other than a metadata generator that Microsoft made for the out-of-process .NET-based Azure Functions (https://github.com/Azure/azure-functions-dotnet-worker/). That stuff is all very preview still, so it could be removed from the product, but it's cool to see Microsoft starting to use it too.

Cheers,

Thomas Bleijendaal
